### PR TITLE
Use Adagrad optimiser for Linear regression by default

### LINKED ISSUE #3210: [BUG] Linear regression model predicts NaN values only
**What is the bug?**
I trained a linear regression model with 5000 features and apparently when calling the `_predict` API only `NaN` values are returned.

I cannot exclude that I'm using parameters that are not ideal and as a consequence lead to the `NaN` predictions. I unsuccessfully tried smaller learning rates but did not experiment with all available parameters and parameter values.

**How can one reproduce the bug?**
1. Get the features at https://gist.github.com/wrigleyDan/a83a5d8294aa0ed493e4feb8cc9d7433
2. Get the notebook to see how I ingest the data, train a model, predict a value: https://gist.github.com/wrigleyDan/16deb9cd8201ec502acda036c0b150b5
3. Run the notebook with the feature data
4. See NaN as the predicted value

**What is the expected behavior?**
The expected behavior is to receive not only `NaN` values but reasonable predictions, in the given example values between 0 and 1.

**What is your host/environment?**
 - OpenSearch v 2.16.0

**Do you have any screenshots?**
See the linked Gist with a notebook example and the data used as features.

**Do you have any additional context?**
Initially reported in the #ml OpenSearch Slack channel: https://opensearch.slack.com/archives/C05BGJ1N264/p1731077205560749

### PR DESCRIPTION
**Pull request by rithin-pullela-aws**
### Description
Updated default optimiser from SIMPLE_SGD to ADA_GRAD

### Related Issues
Resolves #3210

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [ ] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/ml-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by mingshl — COMMENTED** [review]
Hi @rithin-pullela-aws, can you try adding test to verify this change?

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/regression/LinearRegression.java [hunk 1]
```diff
@@ -153,8 +153,8 @@ private void createOptimiser() {
                 optimiser = new RMSProp(learningRate, momentumFactor, epsilon, decayRate);
                 break;
             default:
-                // Use default SGD with a constant learning rate.
-                optimiser = SGD.getSimpleSGD(learningRate, momentumFactor, momentum);
+                // Use AdaGrad by default
+                optimiser = new AdaGrad(learningRate, epsilon);
```
Comment (dhrubo-os) [inline ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/regression/LinearRegression.java:157]: Let's add the issue comment link as a comment here for further reference here.

Comment (rithin-pullela-aws) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/regression/LinearRegression.java:157]: Do you mean as a comment in the code with a link?

Comment (dhrubo-os) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/regression/LinearRegression.java:157]: Yes

**Review by dhrubo-os — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**